framework/SupervisedLearning/ARMA.py

```diff
@@ -254,14 +276,29 @@ def __trainLocal__(self,featureVals,targetVals):
         self._trainingCDF[target] = mathUtils.trainEmpiricalFunction(timeSeriesData, minBins=self._minBins)
       # if this target governs the zero filter, extract it now
       if target == self.zeroFilterTarget:
-        self.notZeroFilterMask = self._trainZeroRemoval(timeSeriesData,tol=self.zeroFilterTol) # where zeros are not
-        self.zeroFilterMask = np.logical_not(self.notZeroFilterMask) # where zeroes are
+        self.notZeroFilterMask = self._trainZeroRemoval(timeSeriesData,tol=self.zeroFilterTol) # where zeros or less than zeros are
+        self.zeroFilterMask = np.logical_not(self.notZeroFilterMask) # where data are
       # if we're removing Fourier signal, do that now.
+      maskRes = timeSeriesData == timeSeriesData
+
+      if target in self.peaks:
+        deltaT=self.pivotParameterValues[-1]-self.pivotParameterValues[0]
+        deltaT=deltaT/(len(self.pivotParameterValues)-1)
+        self.peaks[target]['period']=int(self.peaks[target]['period']/deltaT)
+        for i in range(len(self.peaks[target]['windows'])):
+          self.peaks[target]['windows'][i]['window'][0]=int(self.peaks[target]['windows'][i]['window'][0]/deltaT)
+          self.peaks[target]['windows'][i]['window'][1]=int(self.peaks[target]['windows'][i]['window'][1]/deltaT)
+          self.peaks[target]['windows'][i]['width']=int(self.peaks[target]['windows'][i]['width']/deltaT)
+        groupWin , maskRes=self._peakGroupWindow(timeSeriesData, windowDict=self.peaks[target] )
+        self.peaks[target]['groupWin']=groupWin
+        self.peaks[target]['mask']=maskRes
+
       if target in self.fourierParams:
         self.raiseADebug('... analyzing Fourier signal  for target "{}" ...'.format(target))
         self.fourierResults[target] = self._trainFourier(self.pivotParameterValues,
                                                          self.fourierParams[target],
                                                          timeSeriesData,
+                                                         masks=[maskRes],
```

Comment:
This is fine; I'd just like to toss in a note that in the future we should consider a consolidated masking system for multiple signal processors.

Reply:
Copy that